src/main/java/com/yelp/nrtsearch/server/luceneserver/innerhit/InnerHitContext.java

```diff
@@ -146,6 +146,10 @@ private void validate() {
     }
   }
 
+  public void setSearchContext(SearchContext searchContext) {
+    this.searchContext = searchContext;
```

Comment:
You mean having isExplain in InnertHit and let that to control it? I somewhat feel it is an overkill. And ES just have one explain for the search

Reply:
Agree with the overkill part.
FieldFetchContext is the abstract class of searchContext and innerHitContext. We may define this isExplain here, and set the innerHit's isExplain = searchContext's isExplain.

I personally don't think getSearchContext from the FieldFetchContext is a good idea. It exists because the current fetchTasks need it. In Theory, we can redesign the fetchTasks to get rid of it. Until then, I don't think we shall add more dependencies replying on getSearchContext().

Reply:
TBH the fact that we first create innerHitContext to set in searchContext and then iterate over the innerHitContext again to set searchContext in it makes it look really hacky. I'd recommend passing the explain or the searchRequest as a parameter to `buildInnerHitContext` and specify the explain property via the builder. The innerHitContext will only have the explain property instead of searchContext.

Reply:
updated